docs/spec.md

```diff
@@ -2633,6 +2633,52 @@ planning to address this in
 
 &nbsp;[More Examples](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/interpret/dot_general.mlir)
 
+### dynamic_iota
+
+#### Semantics
+
+Fills an `output` tensor with values in increasing order starting from zero
+along the `iota_dimension` dimension. `output` tensor shape is specified
+dynamically via `output_shape` input. More formally,
+
+`output[result_index] = constant(is_quantized(output) ?
+quantize(result_index[iota_dimension], element_type(output)) :
+result_index[iota_dimension], element_type(output))`.
+
+#### Inputs
+
+| Label | Name             | Type                                                                               | Constraints |
+|-------|------------------|------------------------------------------------------------------------------------|-------------|
+| (I1)  | `iota_dimension` | `si64`                                                                             | (C1)        |
+| (I2)  | `output_shape`   | 1-dimensional tensor constant of type `si64`                                       | (C1), (C2)  |
+
+#### Outputs
+
+| Name     | Type                                                                             | Constraints |
+|----------|----------------------------------------------------------------------------------|-------------|
+| `output` | tensor of integer, floating-point or complex type or per-tensor quantized tensor | (C2)        |
```

Comment:
The name should be [`result`](https://github.com/openxla/stablehlo/blob/205f405613a14554095d678ebd87c1ced924f7af/stablehlo/dialect/StablehloOps.td#L141).

Reply:
This seems to have been copied from `iota`. It seems we mostly use `result`, but a few specs use `output`. We should unify this.

Reply:
Changing name has implications for serialization and we have a ticket to track this https://github.com/openxla/stablehlo/issues/351. Unifying this is out of scope of this PR though.